### PR TITLE
feat: proto for oracle upgrade and its skeleton

### LINKED ISSUE #575: Make proto message for oracle upgrade
Txs
- UpgradeOracle
- ApproveOracleUpgrade

Queries
- OracleUpgradeInfo

### PR DESCRIPTION
**Pull request by H4NLee**
close #575 

proto messages and skeleton code for the txs and queries.

Apart from `OracleRegistration`, I defined `OracleUpgrade` which is a set for `MsgUpgradeOracle` for now. When developing functions, if it seems that it is better to combine them, then let's merge them.

BTW, the name of `OracleUpgrade` is little bit confusing, so if you have any good idea for this, please feel free to leave your thoughts.

### CODE REVIEW

File: x/oracle/types/events.go [hunk 1]
```diff
@@ -3,6 +3,8 @@ package types
 const (
 	EventTypeRegistration              = "oracle_registration"
 	EventTypeApproveOracleRegistration = "approve_oracle_registration"
+	EventTypeUpgrade                   = "oracle_upgrade"
+	EventTypeApproveOracleUpgrade      = "approve_oracle_upgrade"
```
Comment (H4NLee) [inline x/oracle/types/events.go:7]: Also, add separate event for oracle upgrade. This event type will be used in oracle side.

**Comment by H4NLee** [comment]
I merged #578 into this branch and changed base branch to `ft/570/upgrade-endblocker`.
Because both query for oracle upgrade info and proto messages are needed to develop oracle.

**Review by audtlr24 — APPROVED** [review]
lgtm

**Comment by gyuguen** [comment]
I think spliting OracleRegistration and OracleUPgrade is a good choice.
Because required paramter are different for OracleRegistration and OracleUpgrade.
OracleUpgrade is not required oracleCommissionRate, oracleCommissionmaxRate and oracleCommissionMaxChangeRate

**Comment by gyuguen** [comment]
I too think the `OracleUpgrade` name is confusing.
But I can't think of a good name. :)

Let's think about this.

**Review by gyuguen — APPROVED** [review]
LGTM

**Review by inchori — APPROVED** [review]
LGTM!

**Review by youngjoon-lee — COMMENTED** [review]
lgtm

**Pull request merged**